RetailCoder.VBE/UI/Command/ExportAllCommand.cs

```diff
@@ -0,0 +1,77 @@
+using System.IO;
+using System.Windows.Forms;
+using NLog;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+using Rubberduck.Navigation.CodeExplorer;
+using Rubberduck.UI.CodeExplorer.Commands;
+
+namespace Rubberduck.UI.Command
+{
+    [CodeExplorerCommand]
+    public class ExportAllCommand : CommandBase 
+    {
+        private readonly IVBE _vbe;
+        private IFolderBrowserFactory _factory;
+
+        public ExportAllCommand(IVBE vbe, IFolderBrowserFactory folderBrowserFactory) : base(LogManager.GetCurrentClassLogger())
+        {
+            _vbe = vbe;
+            _factory = folderBrowserFactory;
+        }
+
+        protected override bool EvaluateCanExecute(object parameter)
+        {
+            if (parameter == null)
+            {
+                return Evaluate(_vbe.ActiveVBProject);
+            }
+            else if (parameter is CodeExplorerProjectViewModel)
+            {
+                var node = (CodeExplorerProjectViewModel)parameter;
+                return Evaluate((IVBProject)node.Declaration.Project);
```

Comment:
The point is `else if (parameter is CodeExplorerProjectViewModel)` is already soft-casting, so it should be possible to get the `node` without a hard cast. You could have:

    var projectNode = parameter as CodeExplorerProjectViewModel;
    if (projectNode != null)
    {
        return Evaluate(node.Declaration.Project);
    }

    var project = parameter as IVBProject;
    if (project != null)
    {
        return Evaluate(project);
    }

    return false;

Or, more succinctly:

    var projectNode = parameter as CodeExplorerProjectViewModel;
    var project = parameter as IVBProject;
    return Evaluate(projectNode?.Declaration.Project ?? project);

But only if `Evaluate` is modified to return `false` given a `null` reference, i.e.:

    return project != null && !project.IsWrappingNullReference && project.VBComponents.Count > 0;

Note that `Declaration.Project` is exposed as an `IVBProject`, that second hard cast is indeed redundant.